snooty/parser.py

```diff
@@ -424,6 +436,54 @@ def dispatch_departure(self, node: docutils.nodes.Node) -> None:
                 target.children = [identifier]
                 item.term.append(target)
 
+    def handle_tabset(self, node: n.Directive, line: int) -> None:
+        tabset = node.options["tabset"]
+        # retrieve dictionary associated with this specific tabset
+        tab_definitions_list = specparser.SPEC.tabs[tabset]
+        tabid_list: List[str] = []
+
+        for idx, child in enumerate(node.children):
+            if not isinstance(child, n.Directive):
+                self.diagnostics.append(
+                    TabMustBeDirective(str(type(child).__class__.__name__), line)
+                )
+                return
+
+            tabid = child.options["tabid"]
+            if not isinstance(tabid, str):
+                self.diagnostics.append(
+                    UnknownTabID(
+                        tabid,
+                        tabset,
+                        f"{tabid} is of type {str(type(tabid).__class__)}. Tab ids must be strings ",
+                        line,
+                    )
+                )
+                return
+
+            unknown_tabid = True
+            # find matching title given id and insert direcitve_argument
+            for t_idx, entry in enumerate(tab_definitions_list):
+                if entry.id == tabid:
+                    child.argument = [n.Text((line,), entry.title)]
+                    unknown_tabid = False
+                    tabid_list.insert(t_idx, tabid)
+
+            if unknown_tabid:
+                self.diagnostics.append(
+                    UnknownTabID(
+                        tabid,
+                        tabset,
+                        f"{tabid} is not defined in rst.toml for this tabset",
```

Comment:
[nit] Wrong filename:
```suggestion
                        f"{tabid} is not defined in rstspec.toml for this tabset",
```